lib/automate/modules/send.py

```diff
@@ -14,26 +16,55 @@ def __init__(self):
 
     def run(self, to, when, body, sender):
         user, _ = fuzzy.extractOne(sender, SETTINGS["users"].keys())
-        settings = SETTINGS["users"][user]["email"]
-        username = settings.get("username")
-        password = settings.get("password")
-        content = body + f"\n\nRegards,\n{user}"
+        self.settings = SETTINGS["users"][user]["email"]
+        self.username = self.settings.get("username")
+        self.password = self.settings.get("password")
+        self.subject = body.partition("\n")[0]
+        self.content = body + f"\n\nRegards,\n{user}"
 
+        if len(to) == 1:
+            receiver = to[0]
+        elif len(to) == 0:
+            raise NoReceiverError("No receiver was entered")
+        else:
+            raise ToManyReceiversError("Can only handle one (1) receiver at this time")
+
+        if not self.is_email(receiver):
+            # filter out the contacts that does not need to be considered
+            possible_receivers = fuzzy.extract(receiver, SETTINGS["users"].keys())
+            possible_receivers = list(filter(lambda x: x[1] > 75, possible_receivers))
+
+            # if there are multiple possible receivers then return a string of these that will
+            # be displayed to the user.
+            #
+            # The user will then be able to enter a more precise name
+            # that will be sent to the follow-up method
+            if len(possible_receivers) > 1:
+                names = "\n".join(list(map(lambda contact: contact[0], possible_receivers)))
+                return None, "Found multiple contacts: \n" + names + "\nPlease enter the name"
```

Comment:
We will probably need an attribute in the future to keep track of *what* follow up question we are asking. But as you are the first one to use this functionality this isn't needed here.